emcc.py

```diff
@@ -2322,6 +2323,24 @@ def do_binaryen(final, target, asm_target, options, memfile, wasm_binary_target,
           passes.append('minifyWhitespace')
         final = shared.Building.js_optimizer_no_asmjs(final, passes)
       if DEBUG: save_intermediate('postclean', 'js')
+  # replace placeholder strings with correct subresource locations
+  if shared.Settings.SINGLE_FILE:
+    f = open(final, 'r')
+    js = f.read()
+    f.close()
+    f = open(final, 'w')
```

Comment:
The situation of text vs binary mode reads is actually very delicate, and in some places we do need to open and write files in text mode. Those locations should be dependant of `--output_eol linux|windows` mode, or later lead to a pass at the end that changes the line endings, so I think other existing locations do not need to be changed. The test `other.test_output_eol` verifies the line endings, so if that passes, things should be good. It may be useful to add a case for `-s SINGLE_FILE=1` in that test if you are interested. (not urgent though)